source/grid/grid_tools_cache.cc

```diff
@@ -146,6 +152,39 @@ namespace GridTools
       }
     return vertex_kdtree;
   }
+
+
+
+  template <int dim, int spacedim>
+  const RTree<std::pair<BoundingBox<spacedim>, unsigned int>> &
+  Cache<dim, spacedim>::get_covering_rtree() const
+  {
+    if (update_flags & update_covering_rtree)
+      {
+        // TO DO: the locally owned portion of the domain
+        // might consists of more separate pieces: a single
+        // bounding box might not always be the best choice.
+        IteratorFilters::LocallyOwnedCell locally_owned_cell_predicate;
+        const BoundingBox<spacedim>       bbox =
+          GridTools::compute_bounding_box(*tria, locally_owned_cell_predicate);
+
+
+        std::vector<BoundingBox<spacedim>> bbox_v(1, bbox);
+        const auto                         tria_mpi =
+          dynamic_cast<const parallel::Triangulation<dim, spacedim> *>(
+            &(*tria));
+
+        if (tria_mpi != nullptr)
```

Comment:
There is a nice idiom that we use in a number of places for what you are doing here: you can combine lines 173-177 into one statement that makes sure that the `tria_mpi` variable only lives in the `if` branch -- which is exactly where you actually need it. The way to write this then is like this:
```
  if (const auto tria_mpi = dynamic_cast<const parallel::Triangulation<dim, spacedim> *>(&(*tria)))
    ...
```
In other words, you are declaring and initializing a variable in the `if` statement; the condition is then simply whether the variable is non-zero, i.e., whether the pointer is non-`NULL`.

Reply:
That looks like a very nice way of writing it, thank you! Just force-updated the commit